mentoring/mentoring.py

```diff
@@ -235,6 +279,44 @@ def next_step_url(self):
         return '/jump_to_id/{}'.format(self.next_step)
 
     @XBlock.json_handler
+    def get_results(self, queries, suffix=''):
+        results = []
+        if not self.show_extended_feedback():
```

Comment:
@e-kolpakov I agree that the name is not optimal. However giving it a different name will have other, more subtle consequences that are irksome. Chiefly, it will mean that this handler may be called something different from the other functions down the line which it calls, which really do just get results unconditionally. Having the same name here, on the sub-blocks, and in the Javascript makes it easier to follow what's going on, and reduces the number of conditionals needed.

It is my opinion that eventually this function should handle results in the general case, at which point the name will be accurate. I'm adding a docstring to explain this. At the moment, there are two ways for a mentoring block to get its result-- which causes a lot of extra complication, especially in the realm of the results handling above, where we have to decide whether to 'stringify' JSON or not.

The two ways, of course, are either to load the template with a div containing all of the attributes the results page will need, or to feed that template with the data returned from a JSON request instead. It would be better if the JSON request just happens upon load, populating it that way, and then there would only be one way to do it.

However, I believe that is outside of the scope of this task, and also would be too much work to sink into something on the edge of deprecation. It should probably, however, be applied to V2, if something like this hasn't already been done. @antoviaque @bradenmacdonald Thoughts on this?